tests/chainer_tests/functions_tests/math_tests/test_einsum.py

```diff
@@ -113,18 +103,18 @@ def check_forward(self, inputs_data, atol=1e-4, rtol=1e-5):
         out = self.op(*[chainer.Variable(x) for x in inputs_data])
         testing.assert_allclose(self.forward_answer, out.data, atol, rtol)
 
-    @_skip_if_float16
     def test_einsum_forward_cpu(self):
         if self.dtype == numpy.float16:
-            self.check_forward(self.inputs, atol=1e-3, rtol=1e-3)
+            _skip_float16_bug()
+            self.check_forward(self.inputs, atol=2e-3, rtol=1e-3)
         else:
             self.check_forward(self.inputs)
 
     @attr.gpu
     def test_einsum_forward_gpu(self):
         inputs = _tuple_to_gpu(self.inputs)
         if self.dtype == numpy.float16:
-            self.check_forward(inputs, atol=1e-3, rtol=1e-3)
+            self.check_forward(inputs, atol=2e-3, rtol=1e-3)
```

Comment:
Still got quite frequent failures with 1000 repetitions. `atol=5e-3` seems stable. Should we relax it further? Ditto for the NumPy test.